roles/create-openshift-resources/tasks/create_templates.yml

```diff
@@ -0,0 +1,38 @@
+---
+
+- name: Set Default template Facts
+  set_fact:
+    template: "{{ template_item }}"
+
+- name: Fail for Missing name
+  fail: msg="This role requires template.name to be set and non empty"
+  when: template.name is not defined or template.name == ''
+
+- name: "Determine if Objects from {{ template.name }} Exists"
+  command: >
+     {{ openshift.common.client_binary }} get ds,dc,bc,is,svc,pvc,sa,rolebindings -l pbiTemplateName={{ template.name }} -n {{ project.name }} -o json
```

Comment:
in general for automation implementations, I recommend spelling out the abbreviations, such as "dc", "bc", etc. I've seen mistakes made in the past due to not doing do...